draft/0004-user-sign-in-experience.md

```diff
@@ -0,0 +1,453 @@
+| `identified`              | The user has provided a valid identifier and at least one related verification record.         | The user can be identified by the Logto system. The interaction session can access the user's profile and MFA settings for further security checks.       |
+| `verified`                | The user has provide all necessary verification records.                                       | The user is fully verified by the Logto system. The user can proceed to the user profile completion and authentication process based on the SIE settings. |
+| `authenticated`           | The interaction session has been submitted. Authorization granted                              | The user can access the platform resources and services. The interaction session will be terminated once the user logs out or the session expires.        |
+
+### 5.2 Interaction session data
+
+| Field               | Type             | Description                                                                                                                                    |
+| ------------------- | ---------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
+| interactionEvent    | InteractionEvent | he interaction event for the current interaction session.                                                                                      |
+| accountId           | string           | The accountId of the user for the current interaction session. Only available once the user is identified by at least one verified identifier. |
+| verificationRecords | Verification[]   | The verification record list attached to the current interaction session.                                                                      |
+| profile             | Profile          | The user provided profile data in the current interaction session that needs to be stored to user DB.                                          |
+
+## 6. Drawbacks
+
+## 7. Rationale and alternatives
+
+## 8.Future possibilities
+
+## Appendix A. Experience APIs
+
+The experience APIs are a set of APIs that allow end-users to interact with the Logto platform to complete the user interaction flow.
+
+### 1. Interaction events API
+
+```ts
+type Identifier = {
+  type: "username" | "email" | "phone";
+  value: string;
+};
+
+type Verification = {
+  type: "password" | "verification_code";
+  /** Can be omitted if a existing verified verification record exist */
+  value?: string;
+  /** Required if a existing verification record is created for the given identifier */
+  verificationId?: string;
+};
+```
+
+#### Register
+
+`POST /experience/api/register`
```

Comment:
should be `/api/experience/register`?

Reply:
Actually I could not see how these APIs are connected with the interaction lifecycle? E.g. Under what conditions these APIs can be called? How will they affect the interaction status?